app/questionnaire/rules.py

```diff
@@ -185,18 +197,26 @@ def _get_answers_on_routing_path(schema, routing_path, answer_store, repeat_rule
             if answer_id in answer_ids_on_path:
                 repeat_indexes.append(answer_id)
 
-    answers = list(answer_store.filter(answer_ids=repeat_indexes))
+    answers = answer_store.filter(answer_ids=repeat_indexes)
 
     # Exclude answers that are not on the routing path from repeat function
-    answers_to_remove = []
-    for answer in answers:
-        if not _is_answer_on_path(schema, answer, routing_path):
-            answers_to_remove.append(answer)
+    answers_on_path = _get_answers_on_path(answers, schema, routing_path)
+
+    return answers_on_path
+
+
+def _get_answers_on_path(answers, schema, routing_path) -> AnswerStore:
+    """
+    Get any answers that are on the routing path and return an answer store.
+    """
+    answers_to_remove = [answer for answer in answers if not _is_answer_on_path(schema, answer, routing_path)]
+
+    answers_on_path = answers.copy()
 
     for answer in answers_to_remove:
-        answers.remove(answer)
+        answers_on_path.answers.remove(answer)
 
-    return answers
+    return answers_on_path
 
 
 def _is_answer_on_path(schema, answer, routing_path):
```

Comment:
I think it would make more sense if the `_get_answers_on_path` and `_is_answer_on_path` methods were in path_finder.